static/app/components/profiling/flamegraphSearch.tsx

```diff
@@ -45,14 +52,22 @@ function frameSearch(
       for (let i = 0; i < frames.length; i++) {
         const frame = frames[i];
 
-        if (new RegExp(lookup, flags ?? 'g').test(frame.frame.name.trim())) {
-          results[
-            `${
-              frame.frame.name +
-              (frame.frame.file ? frame.frame.file : '') +
-              String(frame.start)
-            }`
-          ] = frame;
+        const re = new RegExp(lookup, flags ?? 'g');
+        const reMatches = Array.from(frame.frame.name.trim().matchAll(re));
+        if (reMatches.length > 0) {
+          const frameId = getFlamegraphFrameSearchId(frame);
+          results[frameId] = {
+            frame,
+            matchIndices: reMatches.reduce((acc, match) => {
```

Comment:
Have you tried by just taking the indice with max distance here? I dont think we necessarily want to draw all matches, if we can find most relevant one thats not something like [93,3] I think it would be enough

Reply:
Yeah, agree. I think finding first match good enough, but we can refine this later like you mentioned.